wrappers/obj-c/ODWLogger.mm

```diff
@@ -14,6 +15,8 @@
 
 using namespace MAT;
 
+static BOOL canUseSDK = YES;
```

Comment:
I'm not an Obj-C dev, so would appreciate some clarity here.

Why is `BOOL` preferred over `bool` or `Boolean` here? As someone reading about this now, it seems weird to have `YES` and `NO` values, instead of the `true`/`false` that is peppered across this SDK.

Reply:
Updated with std::atomic<bool> and true /false